src/components/PDFView/index.tsx

```diff
@@ -0,0 +1,139 @@
+     * Verify that the canvas limits have been calculated already, if not calculate them and put them in Onyx
+     */
+    const retrieveCanvasLimits = () => {
+        if (!maxCanvasArea) {
+            CanvasSize.retrieveMaxCanvasArea();
+        }
+
+        if (!maxCanvasHeight) {
+            CanvasSize.retrieveMaxCanvasHeight();
+        }
+
+        if (!maxCanvasWidth) {
+            CanvasSize.retrieveMaxCanvasWidth();
+        }
+    };
+
+    useEffect(() => {
+        retrieveCanvasLimits();
+        // This rule needs to be applied so that this effect is executed only when the component is mounted
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        // Use window height changes to toggle the keyboard. To maintain keyboard state
+        // on all platforms we also use focus/blur events. So we need to make sure here
+        // that we avoid redundant keyboard toggling.
+        // Minus 100px is needed to make sure that when the internet connection is
+        // disabled in android chrome and a small 'No internet connection' text box appears,
+        // we do not take it as a sign to open the keyboard
+        if (!isKeyboardOpen && windowHeight < prevWindowHeight - 100) {
+            toggleKeyboardOnSmallScreens(true);
+        } else if (isKeyboardOpen && windowHeight > prevWindowHeight) {
+            toggleKeyboardOnSmallScreens(false);
+        }
+    }, [isKeyboardOpen, prevWindowHeight, toggleKeyboardOnSmallScreens, windowHeight]);
+
+    const renderPDFView = () => {
+        const outerContainerStyle = [styles.w100, styles.h100, styles.justifyContentCenter, styles.alignItemsCenter];
+
+        return (
+            <View
+                style={outerContainerStyle}
+                tabIndex={0}
+            >
+                <PDFPreviewer
+                    contentContainerStyle={style as CSSProperties}
+                    file={sourceURL}
+                    pageMaxWidth={variables.pdfPageMaxWidth}
+                    isSmallScreen={isSmallScreenWidth}
+                    maxCanvasWidth={maxCanvasWidth}
+                    maxCanvasHeight={maxCanvasHeight}
+                    maxCanvasArea={maxCanvasArea}
+                    LoadingComponent={<FullScreenLoadingIndicator />}
+                    ErrorComponent={<Text style={errorLabelStyles}>{translate('attachmentView.failedToLoadPDF')}</Text>}
+                    renderPasswordForm={({isPasswordInvalid, onSubmit, onPasswordChange}) => (
+                        <PDFPasswordForm
+                            isFocused={!!isFocused}
+                            isPasswordInvalid={isPasswordInvalid}
+                            onSubmit={onSubmit}
+                            onPasswordUpdated={onPasswordChange}
+                            onPasswordFieldFocused={toggleKeyboardOnSmallScreens}
+                        />
+                    )}
+                />
+            </View>
+        );
+    };
```

Comment:
Shouldn't this be a separate component? This looks like a big calculation each time PDFView rerenders 😄

Reply:
I would avoid this kind of refactor right now, let's focus on the migration only